src/chain/chain.rs

```diff
@@ -996,29 +996,44 @@ impl Chain {
         let is_connected = |target_name: &XorName| connected_peers.contains(&target_name);
 
         let candidates = |target_name: &XorName| {
-            let mut filtered_sections =
+            let filtered_sections =
                 self.closest_sections(target_name)
                     .into_iter()
-                    .map(|(_, members)| {
+                    .map(|(prefix, members)| {
                         (
+                            prefix,
                             members.len(),
                             members.into_iter().filter(is_connected).collect::<Vec<_>>(),
                         )
                     });
-            let best_section = filtered_sections
-                .find(|(len, connected)| connected.len() * 3 >= *len)
-                .ok_or(Error::CannotRoute);
-            best_section.map(|(len, connected)| {
-                (
-                    len,
-                    connected
-                        .into_iter()
-                        .sorted_by(|lhs, rhs| target_name.cmp_distance(lhs, rhs)),
-                )
-            })
+
+            let mut dg_size = 0;
+            let mut node_to_send = Vec::new();
+            for (idx, (prefix, len, connected)) in filtered_sections.enumerate() {
+                node_to_send.extend(connected.into_iter());
+                dg_size = delivery_group_size(len);
+
+                if &prefix == self.our_prefix() {
```

Comment:
just to confirm. So if our section is not the closest (i.e. idx!=0), then we will need to send to all the sections closer to it than us and the self section. But no more than that.
right?
Probably the comment of `Send to all connected nodes so the connected node can forward the message` is bit misleading, as here the nodes that connected but is far than us, doesn't need be to send to ?

Reply:
I could replace node with target.

`Send to all connected targets so they can forward the message`